client/src/App.js

```diff
@@ -103,6 +104,32 @@ function App() {
     setCharOffset([-1, -1])
   }
 
+  // create a function to handle the upload button
+  const handleUploadButtonClick = (event) => {
+    // add multiple files to form data
+    setProgress(0);
+    const formData = new FormData();
+    for (const file of event.target.files) {
+      formData.append('files', file);
+    }
+    //post the files to the server
+    fetch(`${api_endpoint}upload`, {
+      method: 'POST',
+      body: formData
+    })
+    .then(response => response.json())
+    .then(result => {
+      console.log('Success:', result);
+    })
+    .catch(error => {
+      console.error('Error:', error);
+    });
+    // TODO: probably fix this
+    setProgress(50);
+    setTimeout(() => setProgress(75), 3000); // Simulate upload delay
+    setTimeout(() => setProgress(100), 6000); // Simulate upload delay
```

Comment:
This shows 100% completion after 6s even if it's not completed? Might be too misleading. I saw an [onUploadProgress handler](https://www.codingdeft.com/posts/react-upload-file-progress-bar/#when-the-file-is-not-uploaded) in this example that might work here